internal/controller/ols_app_server_deployment.go

```diff
@@ -274,17 +310,19 @@ func (r *OLSConfigReconciler) updateOLSDeployment(ctx context.Context, existingD
 		existingDeployment.Annotations[OLSAppTLSHashKey] != r.stateCache[OLSAppTLSHashStateCacheKey] ||
 		existingDeployment.Annotations[LLMProviderHashKey] != r.stateCache[LLMProviderHashStateCacheKey] {
```

Comment:
@raptorsun If just the AdditionalCAHashStateCacheKey changes ( direct update in the additional CR configmap and  no config CR change, no AppTLS change ,no Provider change  ) , that will trigger a reconcile . During that case  , if deployment exists and we do  updateOLSDeployment - we want to update the annotations with  new AdditionalCAHashStateCacheKey right ? are we missing a case for that ?